projects/cds/src/merchandising/model/strategy.result.ts

```diff
@@ -0,0 +1,23 @@
+export interface MerchandisingProduct {
+  id?: string;
+  name?: string;
+  description?: string;
+  brand?: string;
+  pageUrl?: string;
+  thumbNailImage?: string;
+  mainImage?: string;
+  price?: number;
+  metadata?: Map<string, string>;
```

Comment:
Oh, I didn't know this is a map. As `Map` is not serializable, are there any problems / unusual data structure when sending and / or receiving data?
If it works, then Angular must handle it properly. But bear in mind that ngrx will throw an error if this model class is used for data that's stored to the store. But I believe you wont' be needing the store, so it should be good.

Reply:
Nope, we don't need to use the store then